pkg/tempo/automaticloggingprocessor/automaticloggingprocessor.go

```diff
@@ -0,0 +1,242 @@
+package automaticloggingprocessor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+
+	util "github.com/cortexproject/cortex/pkg/util/log"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/go-logfmt/logfmt"
+	"github.com/grafana/agent/pkg/loki"
+	"github.com/grafana/agent/pkg/tempo/contextkeys"
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/promtail/api"
+	"github.com/prometheus/common/model"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenterror"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+	"go.uber.org/atomic"
+)
+
+const (
+	defaultLokiTag     = "tempologging"
+	defaultServiceKey  = "svc"
+	defaultSpanNameKey = "span"
+	defaultStatusKey   = "status"
+	defaultDurationKey = "dur"
+	defaultTraceIDKey  = "tid"
+
+	typeSpan    = "span"
+	typeRoot    = "root"
+	typeProcess = "process"
+)
+
+type automaticLoggingProcessor struct {
+	nextConsumer consumer.TracesConsumer
+	cfg          *AutomaticLoggingConfig
+	lokiChan     chan<- api.Entry
+	done         atomic.Bool
+
+	logger log.Logger
+}
+
+func newTraceProcessor(nextConsumer consumer.TracesConsumer, cfg *AutomaticLoggingConfig) (component.TracesProcessor, error) {
+	logger := log.With(util.Logger, "component", "tempo automatic logging")
+
+	if nextConsumer == nil {
+		return nil, componenterror.ErrNilNextConsumer
+	}
+
+	if !cfg.Roots && !cfg.Processes && !cfg.Spans {
+		return nil, errors.New("automaticLoggingProcessor requires one of roots, processes, or spans to be enabled")
+	}
+
+	cfg.Overrides.LokiTag = override(cfg.Overrides.LokiTag, defaultLokiTag)
+	cfg.Overrides.ServiceKey = override(cfg.Overrides.ServiceKey, defaultServiceKey)
+	cfg.Overrides.SpanNameKey = override(cfg.Overrides.SpanNameKey, defaultSpanNameKey)
+	cfg.Overrides.StatusKey = override(cfg.Overrides.StatusKey, defaultStatusKey)
+	cfg.Overrides.DurationKey = override(cfg.Overrides.DurationKey, defaultDurationKey)
+	cfg.Overrides.TraceIDKey = override(cfg.Overrides.TraceIDKey, defaultTraceIDKey)
```

Comment:
I think the argument to `override` has to be the non-override values

Reply:
I think this is fine? It needs tests regardless.

Reply:
Yes, sorry. I got confused